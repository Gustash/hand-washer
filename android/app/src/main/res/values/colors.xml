<resources>
  <color name="white">#00A778</color>
</resources>
