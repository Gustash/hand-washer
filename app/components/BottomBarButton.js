import React from 'react'
import { TouchableOpacity, StyleSheet, Text, SafeAreaView } from 'react-native'
import { colors } from 'app/config'

const BottomBarButton = ({
  onPress,
  text,
  style,
  color = colors.lightBlue,
  textColor = colors.white,
}) => (
  <TouchableOpacity onPress={onPress}>
    <SafeAreaView style={[styles.container, { backgroundColor: color }, style]}>
      <Text style={[styles.text, { color: textColor }]}>{text}</Text>
    </SafeAreaView>
  </TouchableOpacity>
)

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    shadowColor: colors.black,
    shadowOpacity: 0.2,
    shadowOffset: {
      height: -4,
    },
    shadowRadius: 16,
  },
  text: {
    fontSize: 17,
    textAlign: 'center',
    marginVertical: 16,
  },
})

export default BottomBarButton
