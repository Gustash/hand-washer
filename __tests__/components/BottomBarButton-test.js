import React from 'react'
import { Text, SafeAreaView } from 'react-native'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import BottomBarButton from '../../app/components/BottomBarButton'
import colors from '../../app/config/colors'

const mockProps = props => ({
  onPress: jest.fn(),
  text: '',
  ...props,
})

describe('<BottomBarButton />', () => {
  let props = mockProps()
  let wrapper = shallow(<BottomBarButton {...props} />)

  it('matches BottomBarButton snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  describe('BottomBarButton applies default props', () => {
    it('defaults background color', () => {
      expect(wrapper.find(SafeAreaView).prop('style')).toEqual(
        expect.arrayContaining([
          expect.objectContaining({
            backgroundColor: colors.lightBlue,
          }),
        ])
      )
    })

    it('defaults text color', () => {
      expect(wrapper.find(Text).prop('style')).toEqual(
        expect.arrayContaining([
          expect.objectContaining({
            color: colors.white,
          }),
        ])
      )
    })
  })

  describe('BottomBarButton applies props', () => {
    beforeAll(() => {
      props = mockProps({
        text: 'Hello World',
        style: {
          marginTop: 1000,
        },
        color: '#FFFFFF',
        textColor: '#000000',
      })
      wrapper = shallow(<BottomBarButton {...props} />)
    })

    it('renders text', () => {
      expect(
        wrapper
          .find(Text)
          .first()
          .prop('children')
      ).toBe('Hello World')
    })

    it('handles onPress', () => {
      wrapper.simulate('press')

      expect(wrapper.prop('onPress')).toHaveBeenCalledTimes(1)
    })

    it('applies style', () => {
      expect(wrapper.find(SafeAreaView).prop('style')).toContainEqual({
        marginTop: 1000,
      })
    })

    it('applies background color', () => {
      expect(wrapper.find(SafeAreaView).prop('style')).toEqual(
        expect.arrayContaining([
          expect.objectContaining({
            backgroundColor: '#FFFFFF',
          }),
        ])
      )
    })

    it('applies text color', () => {
      expect(wrapper.find(Text).prop('style')).toEqual(
        expect.arrayContaining([
          expect.objectContaining({
            color: '#000000',
          }),
        ])
      )
    })
  })
})
