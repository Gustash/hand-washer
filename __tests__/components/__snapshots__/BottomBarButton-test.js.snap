// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomBarButton /> matches BottomBarButton snapshot 1`] = `
<ForwardRef
  onPress={[MockFunction]}
>
  <ForwardRef(SafeAreaView)
    style={
      Array [
        Object {
          "alignItems": "center",
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": -4,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 16,
        },
        Object {
          "backgroundColor": "#00A1D2",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 17,
            "marginVertical": 16,
            "textAlign": "center",
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    />
  </ForwardRef(SafeAreaView)>
</ForwardRef>
`;
